### PR TITLE
CMake: Simplify job limiting

### PR DESCRIPTION
**Pull request by rschu1ze**
Follow-up to #51710

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit 0167ec28cfe113b26cd5b5d9f1d58dada233f0c4 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52196/0167ec28cfe113b26cd5b5d9f1d58dada233f0c4/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: cmake/limit_jobs.cmake [hunk 1]
```diff
@@ -52,20 +43,16 @@ if (CMAKE_BUILD_TYPE_UC STREQUAL "RELWITHDEBINFO" AND ENABLE_THINLTO AND PARALLE
     set (PARALLEL_LINK_JOBS 2)
 endif()
 
-if (PARALLEL_LINK_JOBS AND (NOT NUMBER_OF_LOGICAL_CORES OR PARALLEL_LINK_JOBS LESS NUMBER_OF_LOGICAL_CORES))
+message(STATUS "System has ${NUMBER_OF_LOGICAL_CORES} logical cores and ${TOTAL_PHYSICAL_MEMORY} megabytes of memory. Building with ${PARALLEL_COMPILE_JOBS} compile jobs and ${PARALLEL_COMPILE_JOBS} linker jobs.")
```
Comment (antaljanosbenjamin) [inline cmake/limit_jobs.cmake:46]:
```suggestion
message(STATUS "System has ${NUMBER_OF_LOGICAL_CORES} logical cores and ${TOTAL_PHYSICAL_MEMORY} megabytes of memory. Building with ${PARALLEL_COMPILE_JOBS} compile jobs and ${PARALLEL_LINK_JOBS} linker jobs.")
```

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]
There is one small thing to fix, but apart from that it looks okay.

File: cmake/limit_jobs.cmake (hunk 1 above)
Comment (rschu1ze) [reply cmake/limit_jobs.cmake]: Merci!

**Review by antaljanosbenjamin — APPROVED** [review]

**Pull request merged**